dfastmi/gui/dialog_view.py

```diff
@@ -0,0 +1,595 @@
+from dfastmi.io.RiversObject import RiversObject
+import dfastmi.kernel.core
+
+# View
+import tkinter as tk
+
+class DialogView:
+    _app : QtWidgets.QApplication = None
+    _win : QtWidgets.QMainWindow = None
+    _tabs : QtWidgets.QBoxLayout = None
+    _layout : QtWidgets.QBoxLayout = None
+    _menubar : QtWidgets.QMenuBar = None
+    branch : QtWidgets.QComboBox = None
+    reach : QtWidgets.QComboBox = None
+    qloc : QtWidgets.QLabel = None
+    qthr : QtWidgets.QLineEdit = None
+    ucrit : QtWidgets.QLineEdit = None
+    slength : QtWidgets.QLabel = None
+
+    output_dir : QtWidgets.QLineEdit = None
+    make_plots_edit : QtWidgets.QCheckBox = None
+    save_plots : QtWidgets.QLabel = None
+    save_plots_edit : QtWidgets.QCheckBox = None
+    figure_dir : QtWidgets.QLabel = None
+    figure_dir_edit : QtWidgets.QLineEdit = None
+
+    close_plots : QtWidgets.QLabel = None
+    close_plots_edit : QtWidgets.QCheckBox = None
+
+    def __init__(self, view_model: DialogViewModel):
+        self.view_model = view_model
+        self.root = tk.Tk()
+        self.root.title("D-FAST Morphological Impact")
+        
+        # Initialize GUI components
+        self.create_qt_application()
+        self.create_dialog()
+        self.create_menus()
+        self.create_central_widget()
+        self.create_tabs()
+        self.create_button_bar()
+        # Connect the view model's data_changed signal to update_ui slot
+        self.view_model.branch_changed.connect(self.update_branch)
+        self.view_model.reach_changed.connect(self.update_reach)
+        self.update_qvalues_tabs()
+
+    def update_branch(self, data):
+        self.branch.setCurrentText(data)
+        self.reach.clear()
+        for r in self.view_model.current_branch.reaches:
+            self.reach.addItem(r.name)
+        self.qloc.setText(self.view_model.current_branch.qlocation)
+        self.qthr.setText(str(self.view_model.current_reach.qstagnant))
+        self.ucrit.setText(str(self.view_model.current_reach.ucritical))
+        self.slength.setText(self.view_model.slength)
+        self.reach.setCurrentText(self.view_model.current_reach.name)
+        self.update_qvalues_tabs()
+        self.output_dir.setText(self.view_model.output_dir)
+        self.make_plots_edit.setChecked(self.view_model.plotting)
+        self.save_plots_edit.setChecked(self.view_model.save_plots)
+        self.close_plots_edit.setChecked(self.view_model.close_plots)
+        self.update_condition_files()
+    
+    def update_reach(self, data):
+        self.reach.setCurrentText(data)
+        self.slength.setText(self.view_model.slength)
+    
+    def update_condition_files(self):
+        for condition_tab_index, reference_file in enumerate(self.view_model.reference_files):
+            tab = self._tabs.widget(condition_tab_index +1)
+            if tab:
+                prefix = str(condition_tab_index)+"_"
+                key = prefix +"file1"
```

Comment:
what are these "file1" and "file2" exactly?

Reply:
reference file input and measure file input. But this is original implementation, wanted to update this in a follow up issue
scope is here to refactor to MVVM